crates/optimism/flashblocks/src/convert.rs

```diff
@@ -0,0 +1,182 @@
+//! Conversions from flashblock types to Engine API payload types.
+//!
+//! This module provides the canonical implementations for converting flashblock data structures
+//! into the various versions of execution payloads required by the Engine API.
+//!
+//! TODO: ideally we move this to op-alloy-rpc-types-engine where `from_block_unchecked` is defined
+//! and define a similar called `from_flashblock_sequence_unchecked`.
+//! The goal is to have the conversion logic in one place to avoid missing steps.
+
+use crate::{
+    ExecutionPayloadBaseV1, ExecutionPayloadFlashblockDeltaV1, FlashBlockCompleteSequence,
```

Comment:
@mattsse what's your thoughts on moving above data structures into `op-alloy-rpc-types-engine`?

Currently they're defined in rollup-boost and imported by op-rbuilder, etc.

If we're making reth to support flashblocks natively, it seems reasonable to move those into `op-alloy-rpc-types-engine` and update all other repos to refer to it.

This helps with
1. centralize commonly shared data structures
2. also we can move this convert.rs functionality directly into `op-alloy-rpc-types-engine` and sit alongside with OpExecutionData to share the same block / flashblock -> `ExecutionData` conversion logic

Reply:
> @mattsse what's your thoughts on moving above data structures into op-alloy-rpc-types-engine?

yeah that would be ideal, even if the format changes we can still make that work